File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int preorder(TreeNode* root, int num){
        if(root == NULL) return 0;
        if(root->left == NULL && root->right == NULL) return num * 10 + root->val;
        int l = preorder(root->left, num * 10 + root->val);
        int r = preorder(root->right, num * 10 + root->val);
        return l + r;
    }
    int sumNumbers(TreeNode* root) {
        return preorder(root, 0);
    }
};

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.java
class Solution {
    public int helper(TreeNode root, int value) {
        if(root == null) return 0;

        if(root.left == null && root.right == null) return value * 10 + root.val;

        int l = helper(root.left, value * 10 + root.val);
        int r = helper(root.right, value * 10 + root.val);

        return l + r;
    }
    public int sumNumbers(TreeNode root) {
        return helper(root, 0);
    }
}